### PR TITLE
Run the transaction DryRun locally

### PR DESCRIPTION
**Pull request by steelgeek091**
## Summary

Summary about this PR

- Closes https://github.com/rooch-network/rooch/issues/2685

### CODE REVIEW

File: crates/rooch-rpc-api/src/api/rooch_api.rs [hunk 1]
```diff
@@ -41,9 +41,6 @@ pub trait RoochAPI {
         tx_option: Option<TxOptions>,
     ) -> RpcResult<ExecuteTransactionResponseView>;
 
-    #[method(name = "dryRunRawTransaction")]
-    async fn dry_run(&self, tx_bcs_hex: BytesView) -> RpcResult<DryRunTransactionResponseView>;
-
```
Comment (jolestar) [inline crates/rooch-rpc-api/src/api/rooch_api.rs:46]: We should keep this API for the web DApp.

Comment (steelgeek091) [reply crates/rooch-rpc-api/src/api/rooch_api.rs:46]: OK, i will add the removed API.

File: crates/rooch/src/tx_runner.rs [hunk 2]
```diff
@@ -199,3 +264,90 @@ fn convert_to_verified_tx(
         verified_action,
     ))
 }
+
+pub fn dry_run_tx_locally(
```
Comment (jolestar) [inline crates/rooch/src/tx_runner.rs:268]: Make the `dry_run_tx_locally` and `get_latest_state_root` to async?

File: crates/rooch/src/tx_runner.rs [hunk 3]
```diff
@@ -199,3 +264,90 @@ fn convert_to_verified_tx(
...
+}
+
+fn get_latest_state_root(client: &Client) -> anyhow::Result<Option<Vec<u8>>> {
+    tokio::task::block_in_place(|| {
+        Handle::current().block_on(async {
+            let sequencer_order = client.rooch.sequencer_order().await?;
+            let transaction = client
+                .rooch
+                .get_transactions_by_order(Some(sequencer_order), Some(1), Some(false))
+                .await?;
```
Comment (jolestar) [inline crates/rooch/src/tx_runner.rs:299]: The `rooch_status` RPC function returns the latest state root. We can use the function; we do not need the `sequencer_order` function.

Comment (steelgeek091) [reply crates/rooch/src/tx_runner.rs]: Fixed

File: crates/rooch-open-rpc-spec/schemas/openrpc.json [hunk 4]
```diff
@@ -747,6 +747,20 @@
...
+      "result": {
+        "name": "u64",
+        "required": true,
+        "schema": {
+          "type": "integer",
+          "format": "uint64",
+          "minimum": 0.0
+        }
+      }
+    },
```
Comment (jolestar) [inline crates/rooch-open-rpc-spec/schemas/openrpc.json:763]: Build and update this file

**Review by jolestar — APPROVED** [review]

File: crates/rooch-open-rpc-spec/schemas/openrpc.json (hunk 4 above)
Comment (steelgeek091) [reply crates/rooch-open-rpc-spec/schemas/openrpc.json]: Fixed

**Pull request merged**